docs: fix and improve multiple sections (#4396)
nuxt/framework: docs/content/2.guide/1.concepts/2.vuejs-development.md

# Vue.js Development

> Nuxt uses Vue as a frontend framework and adds features such as components auto-imports and file-based routing. Nuxt 3 integrates Vue 3, the new major release of Vue that enables new patterns for Nuxt users.

Reviewer: ```suggestion
Nuxt uses Vue as a frontend framework and adds features such as component auto-imports and file-based routing. Nuxt 3 integrates Vue 3, the new major release of Vue that enables new patterns for Nuxt users.
```

## Vue.js

> While an in-depth knowledge of Vue is not required to use Nuxt, we recommend that you read the documentation and go through some of the examples on [vuejs.org](https://vuejs.org/)
>

Nuxt has always used Vue as a frontend framework. We chose to build Nuxt on top of Vue for these reasons:

> - The reactivity model of Vue, where a change in the data automatically triggers a change in the interface.

Reviewer: ```suggestion
- The reactivity model of Vue, where a change in data automatically triggers a change in the interface.
```

### Faster rendering

Vue Virtual DOM (VDOM) has been rewritten from the ground up and allows for better rendering performances. On top of that, when working with compiled Single-file components, the Vue compiler can further optimize components at build time by separating static and dynamic markup.

This results in faster first rendering (component creation) and updates, and less memory usage. In Nuxt 3, it enables faster server-side rendering as well.

### Smaller bundle

> With Vue 3 and Nuxt 3, a focus has been put on bundle size reduction. With version 3, most of Vue’s functionality, including template directives and built-in components is tree-shakeable. Your production bundle will not include them if you don’t use them.

Reviewer: ```suggestion
With Vue 3 and Nuxt 3, a focus has been put on bundle size reduction. With version 3, most of Vue’s functionality, including template directives and built-in components, is tree-shakeable. Your production bundle will not include them if you don’t use them.
```

</script>
```

> The [Composition API](https://vuejs.org/guide/extras/composition-api-faq.html) introduced in Vue 3 is not a replacement for the Options API, but enables better logic reuse throughout an application, and a more natural way to group code by concerns in complex components.

Reviewer: ```suggestion
The [Composition API](https://vuejs.org/guide/extras/composition-api-faq.html) introduced in Vue 3 is not a replacement of the Options API, but it enables better logic reuse throughout an application, and is a more natural way to group code by concern in complex components.
```
